### ISSUE TITLE
[BUG] Loading a single Datetime np into a Dataframe fails  

### ISSUE THREAD

**Issue by mlahir1** [rapidsai/cudf#2383]
# Code
---------

Fails
```
import cudf
import numpy as np
from datetime import date

df['b'] = [np.datetime64(date.today())] * 3
df['b'] = np.datetime64(date.today())
```
if we cast it to string, it is successful
```
df['b'] = [np.datetime64(date.today())] * 3
df['b'] = str(np.datetime64(date.today()))
```

# Error Msg
-----------

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/conda/lib/python3.7/site-packages/cudf/dataframe/dataframe.py", line 336, in __setitem__
    self._cols[name] = self._prepare_series_for_add(col)
  File "/conda/lib/python3.7/site-packages/cudf/dataframe/dataframe.py", line 1212, in _prepare_series_for_add
    series = self._sanitize_values(series, SCALAR)
  File "/conda/lib/python3.7/site-packages/cudf/dataframe/dataframe.py", line 1187, in _sanitize_values
    cudautils.gpu_fill_value.forall(arr.size)(arr, col)
  File "/conda/lib/python3.7/site-packages/numba/cuda/compiler.py", line 267, in __call__
    kernel = self.kernel.specialize(*args)
  File "/conda/lib/python3.7/site-packages/numba/cuda/compiler.py", line 817, in specialize
    kernel = self.compile(argtypes)
  File "/conda/lib/python3.7/site-packages/numba/cuda/compiler.py", line 833, in compile
    **self.targetoptions)
  File "/conda/lib/python3.7/site-packages/numba/compiler_lock.py", line 32, in _acquire_compile_lock
    return func(*args, **kwargs)
  File "/conda/lib/python3.7/site-packages/numba/cuda/compiler.py", line 62, in compile_kernel
    cres = compile_cuda(pyfunc, types.void, args, debug=debug, inline=inline)
  File "/conda/lib/python3.7/site-packages/numba/compiler_lock.py", line 32, in _acquire_compile_lock
    return func(*args, **kwargs)
  File "/conda/lib/python3.7/site-packages/numba/cuda/compiler.py", line 51, in compile_cuda
    locals={})
  File "/conda/lib/python3.7/site-packages/numba/compiler.py", line 955, in compile_extra
    return pipeline.compile_extra(func)
  File "/conda/lib/python3.7/site-packages/numba/compiler.py", line 377, in compile_extra
    return self._compile_bytecode()
  File "/conda/lib/python3.7/site-packages/numba/compiler.py", line 886, in _compile_bytecode
    return self._compile_core()
  File "/conda/lib/python3.7/site-packages/numba/compiler.py", line 873, in _compile_core
    res = pm.run(self.status)
  File "/conda/lib/python3.7/site-packages/numba/compiler_lock.py", line 32, in _acquire_compile_lock
    return func(*args, **kwargs)
  File "/conda/lib/python3.7/site-packages/numba/compiler.py", line 254, in run
    raise patched_exception
  File "/conda/lib/python3.7/site-packages/numba/compiler.py", line 245, in run
    stage()
  File "/conda/lib/python3.7/site-packages/numba/compiler.py", line 501, in stage_nopython_frontend
    self.locals)
  File "/conda/lib/python3.7/site-packages/numba/compiler.py", line 1105, in type_inference_stage
    infer.propagate()
  File "/conda/lib/python3.7/site-packages/numba/typeinfer.py", line 915, in propagate
    raise errors[0]
numba.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(<built-in function setitem>) with argument(s) of type(s): (array(datetime64[ms], 1d, A), int32, datetime64[D])
 * parameterized
In definition 0:
    All templates rejected with literals.
In definition 1:
    All templates rejected without literals.
In definition 2:
    All templates rejected with literals.
In definition 3:
    All templates rejected without literals.
In definition 4:
    All templates rejected with literals.
In definition 5:
    All templates rejected without literals.
In definition 6:
    All templates rejected with literals.
In definition 7:
    All templates rejected without literals.
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: typing of setitem at /conda/lib/python3.7/site-packages/cudf/utils/cudautils.py (245)

File "../../conda/lib/python3.7/site-packages/cudf/utils/cudautils.py", line 245:
def gpu_fill_value(data, value):
    <source elided>
    if tid < data.size:
        data[tid] = value
        ^
```

**Comment by trstovall** [comment]
```python
import cudf
import numpy as np
from datetime import date

df = cudf.DataFrame({'a': range(3)})

df['b'] = [np.datetime64(date.today())] * 3    # works!
df['b'] = str(np.datetime64(date.today()))     # works!
df['b'] = np.datetime64(date.today())            # fails!
```

**Comment by randerzander** [comment]
I believe this was resolved by #2401. Please re-open if you still have the issue.
